### DISCUSSION TITLE
[torchmetrics] Difference between dice_score and f1_score

### DISCUSSION THREAD

**Discussion by digital-idiot** [Lightning-AI/torchmetrics#941] [General]
In the `torchmetrics` what is the difference between `f1_score` and `dice_score`? Aren't they theoretically the same metric <sup>[\[1\]](https://chenriang.me/f1-equal-dice-coefficient.html)</sup>? Looking at [source of dice_score](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/functional/classification/dice.py#L62-L113) it appears to be using the formula directly whereas [f1_score](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/functional/classification/f_beta.py#L247-L355) calls `fbeta_score` with `beta=1`.

Also, it is possible to pass `beta` value other than `1` to `f1_score` which is a bit confusing and redundant.

[1] https://chenriang.me/f1-equal-dice-coefficient.html

**Comment by SkafteNicki** [comment]
Hi @digital-idiot,
You are right that `f1_score` and `dice_score` calculates the same thing.
You are also right that `beta` can be set to something else than `1` in `f1_score`, which is a mistake.
We are going to do a major refactor of the classification package in the next release where we will probably remove `dice_score` (as `f1_score` is more general) and rectify the other mistake.
Thanks for bringing this to our attention :]
